src/cephadm/cephadmlib/templating.py

```diff
@@ -17,6 +17,8 @@ class Templates(str, enum.Enum):
 
     ceph_service = 'ceph.service.j2'
     agent_service = 'agent.service.j2'
+    cluster_logrotate_config = 'cluster.logrotate.config.j2'
+    cephadm_logrotate_config = 'cephadm.logrotate.config.j2'
```

Comment:
this seems to have caused a test failure, interestingly in a test where it runs `cephadm -v add-repo --release octopus`
```
2023-11-08T09:57:10.619 INFO:tasks.workunit.client.0.smithi174.stderr:+ sudo tmp.test_repos.sh.6K4kO9/tmp.cephadm.u8QPlK -v add-repo --release octopus
2023-11-08T09:57:10.758 INFO:tasks.workunit.client.0.smithi174.stderr:Traceback (most recent call last):
2023-11-08T09:57:10.758 INFO:tasks.workunit.client.0.smithi174.stderr:  File "<frozen zipimport>", line 177, in get_data
2023-11-08T09:57:10.759 INFO:tasks.workunit.client.0.smithi174.stderr:KeyError: '/home/ubuntu/cephtest/mnt.0/client.0/tmp/tmp.test_repos.sh.6K4kO9/tmp.cephadm.u8QPlK/cephadmlib/templates/cephadm.logrotate.config.j2'
2023-11-08T09:57:10.759 INFO:tasks.workunit.client.0.smithi174.stderr:
2023-11-08T09:57:10.759 INFO:tasks.workunit.client.0.smithi174.stderr:During handling of the above exception, another exception occurred:
2023-11-08T09:57:10.759 INFO:tasks.workunit.client.0.smithi174.stderr:
2023-11-08T09:57:10.759 INFO:tasks.workunit.client.0.smithi174.stderr:Traceback (most recent call last):
2023-11-08T09:57:10.760 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/jinja2/loaders.py", line 357, in get_source
2023-11-08T09:57:10.760 INFO:tasks.workunit.client.0.smithi174.stderr:  File "<frozen zipimport>", line 179, in get_data
2023-11-08T09:57:10.760 INFO:tasks.workunit.client.0.smithi174.stderr:OSError: [Errno 0] : '/home/ubuntu/cephtest/mnt.0/client.0/tmp/tmp.test_repos.sh.6K4kO9/tmp.cephadm.u8QPlK/cephadmlib/templates/cephadm.logrotate.config.j2'
2023-11-08T09:57:10.760 INFO:tasks.workunit.client.0.smithi174.stderr:
2023-11-08T09:57:10.760 INFO:tasks.workunit.client.0.smithi174.stderr:The above exception was the direct cause of the following exception:
2023-11-08T09:57:10.761 INFO:tasks.workunit.client.0.smithi174.stderr:
2023-11-08T09:57:10.761 INFO:tasks.workunit.client.0.smithi174.stderr:Traceback (most recent call last):
2023-11-08T09:57:10.761 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
2023-11-08T09:57:10.761 INFO:tasks.workunit.client.0.smithi174.stderr:    return _run_code(code, main_globals, None,
2023-11-08T09:57:10.761 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
2023-11-08T09:57:10.761 INFO:tasks.workunit.client.0.smithi174.stderr:    exec(code, run_globals)
2023-11-08T09:57:10.762 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/__main__.py", line 7613, in <module>
2023-11-08T09:57:10.762 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/__main__.py", line 7587, in main
2023-11-08T09:57:10.762 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/cephadmlib/logging.py", line 217, in cephadm_init_logging
2023-11-08T09:57:10.762 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/cephadmlib/logging.py", line 237, in write_cephadm_logrotate_config
2023-11-08T09:57:10.763 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/cephadmlib/templating.py", line 58, in render
2023-11-08T09:57:10.763 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/jinja2/environment.py", line 1010, in get_template
2023-11-08T09:57:10.763 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/jinja2/environment.py", line 969, in _load_template
2023-11-08T09:57:10.763 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/jinja2/loaders.py", line 126, in load
2023-11-08T09:57:10.763 INFO:tasks.workunit.client.0.smithi174.stderr:  File "/tmp/tmp3z2iidpq.cephadm.build/jinja2/loaders.py", line 359, in get_source
2023-11-08T09:57:10.763 INFO:tasks.workunit.client.0.smithi174.stderr:jinja2.exceptions.TemplateNotFound: cephadm.logrotate.config.j2
```

Reply:
I checked out your pr and built the zipapp locally, and tried this out on a few distros. I didn't hit this traceback. It might help to examine what build created this zipapp, and perhaps examine the zip itself.

Reply:
this seemed to be an issue with using the compiled binary with relative paths after the jinja2 addition. Fix is https://github.com/ceph/ceph/pull/54487

Reply:
to reliability reproduce this issue, I actually needed to remove the cephadm logrotate file during each test iteration (since any subsequent invocation after would actually succeed)

something like this:
```
$ rm /etc/logrotate.d/cephadm && cephadm list-networks # failure
$ cephadm list-networks # success after a second attempt
```

I've also confirmed [#54487](https://github.com/ceph/ceph/pull/54487#discussion_r1396066366) appears to fix the issues seen in this PR